src/Data/Conduit/Tar.hs

```diff
@@ -270,3 +245,482 @@ withEntries :: MonadThrow m
+                    else tarPayload size' header cont
+            Nothing -> throwM $ TarCreationError "Stream finished abruptly. Not enough payload."
+
+
+
+tarHeader :: MonadThrow m =>
+             FileOffset -> ConduitM (Either Header ByteString) ByteString m FileOffset
+tarHeader offset = do
+    eContent <- await
+    case eContent of
+        Just c@(Right _) -> do
+            leftover c
+            throwM $ TarCreationError "Received payload without a corresponding Header."
+        Just (Left header) -> do
+            packHeader header >>= yield
+            tarPayload 0 header tarHeader
+        Nothing -> do
+            yield terminatorBlock
+            return $ offset + fromIntegral (S.length terminatorBlock)
+
+
+
+tarFileInfo :: MonadThrow m =>
+               FileOffset -> ConduitM (Either FileInfo ByteString) ByteString m FileOffset
+tarFileInfo offset = do
+    eContent <- await
+    case eContent of
+        Just (Right _) ->
+            throwM $ TarCreationError "Received payload without a corresponding FileInfo."
+        Just (Left fi) -> do
+            eHeader <- headerFromFileInfo offset fi
+            case eHeader of
+                Left (FileNameTooLong _) -> do
+                    let fPath = filePath fi
+                        fPathLen = fromIntegral (S.length fPath + 1)
+                        pad =
+                            case fPathLen `mod` blockSize of
+                                0 -> 0
+                                x -> blockSize - x
+                    eHeader' <-
+                        headerFromFileInfo
+                            (offset + blockSize + fPathLen + pad)
+                            (fi {filePath = S.take 100 fPath})
+                    header <- either throwM return eHeader'
+                    pHeader <- packHeader header
+                    pFileNameHeader <-
+                        packHeader $
+                        (defHeader offset)
+                        { headerFileNameSuffix = "././@LongLink"
+                        , headerPayloadSize = fPathLen
+                        , headerLinkIndicator = 76 -- 'L'
+                        , headerMagicVersion = gnuTarMagicVersion
+                        }
+                    yield pFileNameHeader
+                    yield fPath
+                    yield $ S.replicate (fromIntegral pad + 1) 0
+                    yield pHeader
+                    tarPayload 0 header tarFileInfo
+                Left exc -> throwM exc
+                Right header -> do
+                    packHeader header >>= yield
+                    tarPayload 0 header tarFileInfo
+        Nothing -> return offset
+
+
+
+
+
+-- | Create a tar archive by suppying `FileInfo`, which must be immediately
+-- followed by the file content, whenever its type is `FTNormal`.
+tar :: MonadResource m =>
+       ConduitM (Either FileInfo ByteString) ByteString m FileOffset
+tar = do
```

Comment:
For `tar` and `tarEntries`: are they handling a single entry, or the entire stream of entries? It's hard understand that from the types and the docs.

Reply:
Both functions expect a stream of Entries/FileInfos. Does wording it that way clarifies it a bit?
```
-- | Create a tar archive by suppying a stream of `Left` `FileInfo`s. Whenever a
-- file type is `FTNormal`, it must be immediately followed by its content as
-- `Right` `ByteString`. The produced `ByteString` is in the raw tar format and
-- is properly terminated at the end, therefore it should no be modified
-- afterwards.
```

Reply:
Yeah, looks good. I think I may have just been having trouble following the codepaths properly.